### PR TITLE
[convertutf] Add package

### PR DESCRIPTION
**Pull request by Qudix**
Add [`convertutf`](https://github.com/xmake-mirror/convertutf)

Pre-requisite for `simpleini`

### CODE REVIEW

File: packages/c/convertutf/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    add_versions("v1.0.1", "ecd79378cc9f24034a2edebf5f198e84c9eb8046261f8208908d2b29d4fa4d99")
+
+    on_install(function (package)
+        io.writefile("xmake.lua", [[
+            target("convertutf")
+                set_kind("$(kind)")
+                set_languages("c99")
+                add_files("ConvertUTF.c")
+                add_headerfiles("ConvertUTF.h")
+        ]])
```
Comment (waruqi) [inline packages/c/convertutf/xmake.lua:16]: please add shared/static

https://github.com/xmake-io/xmake-repo/blob/47a1bd8a326ba23202adfb78797f2d3b115665f6/packages/z/zlib/xmake.lua#L66

Comment (Qudix) [reply packages/c/convertutf/xmake.lua:16]: I don't believe it exports anything so it may be best to make it static only

Comment (waruqi) [reply packages/c/convertutf/xmake.lua:16]: We can force to export it. see 

https://github.com/xmake-io/xmake-repo/blob/47a1bd8a326ba23202adfb78797f2d3b115665f6/packages/l/libimagequant/xmake.lua#L21-L23

Comment (Qudix) [reply packages/c/convertutf/xmake.lua:16]: Oh, interesting. I'll give that a shot.

File: packages/c/convertutf/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    add_versions("v1.0.1", "ecd79378cc9f24034a2edebf5f198e84c9eb8046261f8208908d2b29d4fa4d99")
+
+    on_install(function (package)
+        io.writefile("xmake.lua", [[
+            target("convertutf")
+                set_kind("$(kind)")
+                set_languages("c99")
+                add_files("ConvertUTF.c")
+                add_headerfiles("ConvertUTF.h")
+                if is_kind("shared") then
```
Comment (waruqi) [inline packages/c/convertutf/xmake.lua:16]: add `is_plat("windows")`

we need only export it for windows.

**Pull request merged**